components/file_upload/file_upload.jsx

```diff
@@ -68,17 +120,17 @@ class FileUpload extends React.Component {
         this.setState({requests});
     }
 
-    fileUploadFail(clientId, channelId, err) {
+    fileUploadFail = (clientId, channelId, err) => {
         this.props.onUploadError(err, clientId, channelId);
     }
 
-    uploadFiles(files) {
+    uploadFiles = (files) => {
```

Comment:
Please find similar implementation at `mattermost-mobile`.

It might need to add `form-data` in `package.json` and the `import FormData from 'form-data';` here